src/Sylius/Behat/spec/Context/Domain/ProductContextSpec.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Behat\Context\Domain;
+
+use Behat\Behat\Context\Context;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class ProductContextSpec extends ObjectBehavior
+{
+    function let(
+        SharedStorageInterface $sharedStorage,
+        RepositoryInterface $productRepository,
+        RepositoryInterface $variantRepository,
+        RepositoryInterface $reviewRepository
+    ) {
+        $this->beConstructedWith($sharedStorage, $productRepository, $variantRepository, $reviewRepository);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Behat\Context\Domain\ProductContext');
+    }
+
+    function it_implements_context_interface()
+    {
+        $this->shouldImplement(Context::class);
+    }
+
+    function it_deletes_a_product(
+        $sharedStorage,
+        $productRepository,
+        ProductInterface $product
+    ) {
+        $productRepository->findOneBy(['name' => 'Model'])->willReturn($product);
+
+        $sharedStorage->set('deleted_product', $product)->shouldBeCalled();
+        $productRepository->remove($product)->shouldBeCalled();
+
+        $this->iDeleteTheProduct('Model');
+    }
+
+    function it_checks_if_there_is_no_variant($variantRepository)
```

Comment:
We need checks that asserts throwing exceptions too. Otherwise, empty brand new generated function would pass this specs